src/dbami/db.py

```diff
@@ -406,20 +402,85 @@ async def _update_schema_version(
             await self.execute_sql(schema_sql, conn=conn)
             await self.execute_sql(table_sql, conn=conn)
 
+    async def _update_schema_version(
+        self,
+        version: int,
+        conn: asyncpg.Connection,
+    ) -> None:
+        await self._create_schema_and_version_table(conn)
+
+        version_sql, params = render(
+            "INSERT INTO :version_table (version) VALUES (:version)",
+            version_table=V(self.schema_version_table),
+            version=version,
+        )
+
         await self.execute_sql(version_sql, *params, conn=conn)
 
+    @asynccontextmanager
+    async def migration_lock(
+        self,
+        use_lock: bool = True,
+        timeout_ms: int = 0,
+        **kwargs,
+    ) -> AsyncIterator[asyncpg.Connection]:
+        async with self.get_db_connection(**kwargs) as conn:
+            if not use_lock:
+                yield conn
+                return
+
+            await self._create_schema_and_version_table(conn)
```

Comment:
1. Move this to the create database method.
2. Catch error (L463) when executing the lock query if the table does not exist to provide useful error message to user.

Reply:
After tentatively implementing this locally and some offline discussions, it was decided to simplify the database adivsory lock by not integrating the schema version table object id into the lock; i.e., we will use a single arbitrary integer for the lock id.